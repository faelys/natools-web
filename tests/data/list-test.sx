(elements
   (title "List Test")
   (main-text
      (comments
         "<p>Empty list:"
         (list (limit 2) (ellipsis " …") (template " |"))
         "</p>\n")
      (tags tags
         "<p>Template list:"
         (all-descendants (template " " (name (-1))))
         "</p>\n<p>Reverse:"
         (all-descendants reverse (template " " (name (-1))))
         "</p>\n<p>Beginning:"
         (all-descendants (limit 2) (ellipsis " …") (template " " (name (-1))))
         "</p>\n<p>Reverse beginning:"
         (all-descendants (reverse) (limit 2)
            (ellipsis-prefix "[") (ellipsis-suffix "]")
            (template " " (name (-1))))
         "</p>\n<p>Tight limit:"
         (all-descendants (limit 5) (ellipsis " …") (template " " (name (-1))))
         "</p>\n"))
   (sidebar ""))
(comments (backend comments list-test))
