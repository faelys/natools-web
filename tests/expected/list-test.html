<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
  <head>
    <title>List Test</title>
  </head>
  <body>
<h1>List Test</h1>
<p>Empty list:</p>
<p>Template list: topic1 other-subtopic subtopic topic2 topic3</p>
<p>Reverse: topic3 topic2 subtopic other-subtopic topic1</p>
<p>Beginning: topic1 other-subtopic â€¦</p>
<p>Reverse beginning:[ topic3 topic2]</p>
<p>Tight limit: topic1 other-subtopic subtopic topic2 topic3</p>
    <div class="sidebar">
    </div>
    <div class="footer">
<p>Default footer: <a href="/index">return to index</a></p>
    </div>
  </body>
</html>
